### ISSUE TITLE
a11y Date-picker component is unusable in high contrast

### ISSUE THREAD

**Issue by mpriour** [Esri/calcite-components#4005]
### Summary

In high contrast mode on Windows, the date-picker component is not useable.
* There is no visible indicator of the selected date.
* There are too many interior borders.
* There is no visible date range indicator.
* There is no visible indicator of a keyboard focused date.

This issue was discovered as part of #3887 - [High contrast audit](https://github.com/Esri/calcite-components/wiki/High-Contrast-Audit#datepicker) and identified as a severity 4 out of 4 issue. 

### Actual Behavior

Here is a date-picker with a selected date range in high contrast mode on Windows

<img height="300" alt="date-picker-high-contrast" src="https://user-images.githubusercontent.com/142636/149594591-be01da90-a63c-43ab-b9b9-74b06c92e9dd.png">

### Expected Behavior

It should be the high contrast equivalent to this:

<img height="300" alt="date-picker-normal" src="https://user-images.githubusercontent.com/142636/149594572-6d23e6b4-1510-4b49-a982-885fd43ed8e6.png">

There are some open questions about what the high contrast version of this is, since not all the styles and features directly translate to a high contrast rendering pattern.

### Reproduction Sample

https://esri.github.io/calcite-components/?path=/story/components-controls-datepicker--range

### Reproduction Steps

1.  Set the theme to a high contrast accessibility theme on a Windows machine or VM.
2. Open the above sample.
3. Attempt to interact with it via mouse or keyboard.
4. Observe the issues stated above.

### Reproduction Version

@esri/calcite-components@1.0.0-latest

### Relevant Info

OS: Windows 7, 10, or 11 with a high contrast accessibility theme selected
Browser: any (Chrome, Edge, Firefox)

### Source

- [X] CDN
- [X] NPM package

**Comment by mpriour** [comment]
This issue could use some design help, @macandcheese. It might also benefit from some input by an actual high contrast user like @MehulChoksey.

By moving the `outline-none` tailwind apply rules to the focus state, it solves the major problems of the single date date-picker.
<img width="550" alt="date-picker with better focus rules" src="https://user-images.githubusercontent.com/142636/151053020-1339602d-3227-49b2-8ac0-9b1aa8cd749c.png">

You can see the selected date on the right hand component and the keyboard focused date (in green) on the left hand component in the screenshot above.

However, the date range use case is harder. The number one question would be should the selected range be solid or just outlines. Here are some rough examples (this use case has a lot of classes that potentially need adjustment)

**Filled date range**
<img width="250" alt="filled date range in a date-picker" src="https://user-images.githubusercontent.com/142636/151053598-7b45044c-6e73-46bb-91e8-9c80dbd02380.png">

**Outlined date range**
<img width="250" alt="outlined date range in a date-picker" src="https://user-images.githubusercontent.com/142636/151053643-29d313b3-6486-4322-a6e1-4b30f6ba2d0a.png">

The second question is what color to use to indicate the date range. Should it use the same color as the component border and selected date(s) or the focus / highlight color as in the examples above, or a different color type?

**Comment by macandcheese** [comment]
It seems like we could simplify that and use the secondary color (like in your filled example), which matches the approach on switch, for example. I'm not sure if that introduces further complexity to ensure the text color of numbers works.

If the rounded "ends" cause issues, it may be simpler to use a full fill on this. Perhaps we could look at a few other design systems examples for inspiration to see how they handle high contrast.

**Comment by mpriour** [comment]
@macandcheese I'm currently writing an article about high contrast support in other design systems, basically it is rarely considered or designed for.

The text in the filled example is surrounded in a rectangle of black (the window background color). This ensures that the text is always readable no matter what color it is on top of. That is a W3C standard that all the browsers are doing in high contrast mode.

I'll do a bit more with the CSS and propose a design based on your feedback.

**Comment by MehulChoksey** [comment]
My preference will be ![image](https://user-images.githubusercontent.com/854529/151065324-8f291ff8-8637-4ba5-85f2-aae77aada368.png) . Start and end cell background could be the green if possible (same color as that for the other cells in the range)

**Comment by mpriour** [comment]
@macandcheese & @MehulChoksey here are some screen shots with the `highlight` versus `canvasText` colors for the fill. I couldn't easily get the fill to draw in the start and end dates, but I could remove the fill from there, so that is what these show.

|Fill style|Selected|Selecting end date|Adjusting start date|
|-|-|-|-|
|`highlight`|<img width="248" alt="Screen Shot 2022-01-25 at 4 20 24 PM" src="https://user-images.githubusercontent.com/142636/151072975-0dd9ed7c-e33c-4329-9c18-8c1bc764f7fc.png">|<img width="248" alt="Screen Shot 2022-01-25 at 4 26 57 PM" src="https://user-images.githubusercontent.com/142636/151073007-e6dce4bc-4112-4bbd-9be8-19b31d6cc1d8.png">|<img width="248" alt="Screen Shot 2022-01-25 at 4 26 44 PM" src="https://user-images.githubusercontent.com/142636/151073024-9c3b2a5f-0857-40b7-94ce-7e6d24e33478.png">|
|`canvasText`|<img width="248" alt="Screen Shot 2022-01-25 at 4 22 10 PM" src="https://user-images.githubusercontent.com/142636/151073116-c08a8828-2705-4bdb-9ed2-52b1354da89c.png">|<img width="248" alt="Screen Shot 2022-01-25 at 4 23 11 PM" src="https://user-images.githubusercontent.com/142636/151073136-e4703cd0-28a1-4034-9617-f7d50f1aed55.png">|<img width="248" alt="Screen Shot 2022-01-25 at 4 22 49 PM" src="https://user-images.githubusercontent.com/142636/151073158-f7b9538e-aca8-44cc-b081-e73e3a94c3bf.png">|

I think using the text color for the filled range selection actually causes readability issues with the numbers and using the highlight color is a better option.

**Comment by MehulChoksey** [comment]
Agree. Like the highlight one better...
Also, the color contrast for Next and Prev btn for the Month selection could be a bit more...

**Comment by macandcheese** [comment]
> Agree. Like the highlight one better...
Agreed

**Comment by macandcheese** [comment]
> Agree. Like the highlight one better...


Agreed.

**Comment by benelan** [comment]
verified on beta.77
